core/src/main/java/org/elasticsearch/action/ingest/SimulatePipelineTransportAction.java

```diff
@@ -56,6 +57,10 @@ protected void doExecute(SimulatePipelineRequest request, ActionListener<Simulat
             } else {
                 simulateRequest = SimulatePipelineRequest.parse(source, request.isVerbose(), pipelineStore);
             }
+        } catch (PipelineFactoryError e) {
+            // TODO(talevy): wrap SimulatePipelineResponse & PipelineFactoryErrorActionResponse into ONE object so this would work
+            listener.onResponse(new PipelineFactoryErrorActionResponse(e));
```

Comment:
yeah, that would work. I don't know why, but yesterday I felt like having this field that is not always applicable to be really confusing. but it would definitely work and be less refactoring